src/transaction/locator_sr.c

```diff
@@ -13937,6 +13941,166 @@ has_errors_filtered_for_insert (std::vector<int> error_filter_array)
+{
+  assert (classname != nullptr);
+
+  if (csect_enter (thread_p, CSECT_LOCATOR_SR_CLASSNAME_TABLE, INF_WAIT) != NO_ERROR)
+    {
+      assert (false);
+      return ER_FAILED;
+    }
+
+  if (csect_enter (thread_p, CSECT_CT_OID_TABLE, INF_WAIT) != NO_ERROR)
+    {
+      assert (false);
+      csect_exit (thread_p, CSECT_LOCATOR_SR_CLASSNAME_TABLE);
+      return ER_FAILED;
+    }
+
+  // *INDENT-OFF*
+  scope_exit <std::function <void (void)>> classname_csect_exit ([&thread_p, &classname] ()
+      {
+        csect_exit (thread_p, CSECT_CT_OID_TABLE);
+        csect_exit (thread_p, CSECT_LOCATOR_SR_CLASSNAME_TABLE);
+      });
+  // *INDENT-ON*
+
+  LOCATOR_CLASSNAME_ENTRY *const entry = (LOCATOR_CLASSNAME_ENTRY *) mht_get (locator_Mht_classnames, classname);
+  if (entry == nullptr)
+    {
+      assert (false);
+      return ER_FAILED;
+    }
+
+  (void) locator_force_drop_class_name_entry (entry->e_name, entry, NULL);
+
+  return NO_ERROR;
+}
+
+/* locator_update_classname_entry () - Update an entry from the locator_Mht_classnames.
+ *                                     It remove the old entry (old_classname) and insert the new one (new_classname).
+ *                                     used in PTS replicator (atomic_replicator)
+ * return	      : error code
+ * thread_p (in)      : thread entry
+ * old_classname (in) : old classname to be removed
+ * new_classname (in) : new classname to be inserted
+ */
+int
+locator_update_classname_entry (THREAD_ENTRY * thread_p, const char *old_classname, const char *new_classname)
+{
+  assert (old_classname != nullptr);
+  assert (new_classname != nullptr);
+
+  LOCATOR_CLASSNAME_ENTRY *new_entry = nullptr;
+
+  if (csect_enter (thread_p, CSECT_LOCATOR_SR_CLASSNAME_TABLE, INF_WAIT) != NO_ERROR)
+    {
+      assert (false);
+      return ER_FAILED;
+    }
+
+  if (csect_enter (thread_p, CSECT_CT_OID_TABLE, INF_WAIT) != NO_ERROR)
+    {
+      assert (false);
+      csect_exit (thread_p, CSECT_LOCATOR_SR_CLASSNAME_TABLE);
+
+      return ER_FAILED;
+    }
+
+  // *INDENT-OFF*
+  scope_exit <std::function <void (void)>> classname_csect_exit ([&thread_p] ()
+      {
+        csect_exit (thread_p, CSECT_CT_OID_TABLE);
+        csect_exit (thread_p, CSECT_LOCATOR_SR_CLASSNAME_TABLE);
+      });
+  // *INDENT-ON*
+
+  LOCATOR_CLASSNAME_ENTRY *const old_entry =
+    (LOCATOR_CLASSNAME_ENTRY *) mht_get (locator_Mht_classnames, old_classname);
+  assert (old_entry != nullptr);
+
+  /* create new entry with new key (new_classname) */
+  new_entry = (LOCATOR_CLASSNAME_ENTRY *) malloc (sizeof (*new_entry));
+  if (new_entry == nullptr)
+    {
+      er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_OUT_OF_VIRTUAL_MEMORY, 1, sizeof (*new_entry));
+      return ER_OUT_OF_VIRTUAL_MEMORY;
+    }
+
+  locator_initialize_classname_entry (new_entry, new_classname, &old_entry->e_current.oid);
+
+  (void) mht_put (locator_Mht_classnames, new_entry->e_name, new_entry);
```

Comment:
Can we add an assert that the entry doesn't exist?

Reply:
Yes! I will add